src/Autohide.js

```diff
@@ -182,10 +186,55 @@ define(function (require, exports, module) {
         exposeDelay = delay;
     }
 
+    function showDelayDialog() {
+        var Dialog = Dialogs.showModalDialog(
+            brackets.DIALOG_ID_SAVE_CLOSE,
+            Strings.AUTOHIDE_DELAY_MODAL_TITLE,
+            Mustache.render(sliderTemplate, {
+                exposeDelay: exposeDelay,
+                millisecondsLabel: Strings.AUTOHIDE_DELAY_MODAL_MS_LABEL
+            })
+        );
+
+        var $slider = $("#outline-delay-slider");
+        var $inputbox = $("#outline-delay-inputbox");
+        var inputboxCurrentValue = $inputbox.val();
+
+        Dialog.done(function (buttonId) {
+            if (buttonId === "ok") {
+                prefs.set("autohideDelay", $slider.val());
+            }
+        });
+
+        // Slider change handler
+        $slider.on("input", function () {
+            inputboxCurrentValue = $slider.val();
+            $inputbox.val(inputboxCurrentValue);
+        });
+
+        // Inputbox handler
+        $inputbox.keyup(function (ev) {
+            if ((ev.keyCode < 48 || ev.keyCode > 57) && ev.keyCode !== 8) {
+                $(this).val(parseInt(inputboxCurrentValue, 10));
+            } else if ($(this).val() < 0 || $(this).val() > 1000) {
+                $(this).val(parseInt(inputboxCurrentValue, 10));
+            } else {
+                if ($(this).val() === "") {
+                    $(this).val(0);
```

Comment:
I'm not sure about this behavior. It feels a bit irritating. Imagine the workflow:
Current value `300`, want to change to `500`. Focus on input, delete everything and want to type `500`, but the input is never empty. It still works (due to parsing ints), but it doesn't look like it.

Instead, maybe leave the input empty, use `placeholder="0"` in the HTML and treat empty as 0 when parsing.

Reply:
Take a look at this and let me know what you think, please.
```
        // Inputbox handler on focus
        $inputbox.focus(function () {
            $(this).select();
        });

        // Inputbox handler to prevent not numeric or backspace input
        $inputbox.keypress(function (ev) {
            // The numbers (0-9) on the keyboard correspond to this range of key codes (48-57).
            // Key code 8 correspond to backspace key.
            // Only these eleven keys are allowed.
            if ((ev.keyCode < 48 || ev.keyCode > 57) && ev.keyCode !== 8) {
                ev.preventDefault();
            }
        });

        // Input handler to ensure a valid value in any case
        $inputbox.keyup(function () {
            var parsed = parseInt($(this).val(), 10);
            if (!$.isNumeric(parsed)) {
                parsed = 0;
            }
            if (parsed > 1000) {
                currentValue = 1000;
                $(this).val(currentValue);
            } else {
                currentValue = parsed;
            }
            $slider.val(currentValue);
        });
```

Reply:
Yes, I think that looks better.

Can the key validation be done by using [`type="number"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number) on the input?